theano/tensor/elemwise.py

```diff
@@ -1059,35 +1076,110 @@ def _c_all(self, node, nodename, inames, onames, sub):
                                      fortran=alloc_fortran)
             alloc += cgen.make_checks([list(range(nnested))], [odtype],
                                       dict(sub, lv0=oname))
-        olv_index = i  # index of the last output
+        # index of the last output
```

Comment:
I'm not sure I understand.

If I'm not mistaken the allocated outputs are already only the "real" outputs not present in the destroy map.

Line 1014 of this file reads : 
        # These are the outputs that we will need to allocate
        # (output, name, name of the c type), transposed
        real = list(zip(*[(r, s, r.type.dtype_specs()[1])
                          for r, s in izip(node.outputs, onames)
                          if r not in dmap]))

to declare the array then used line 1068 to allocate the outputs. From my understanding of this, no allocated outputs are freed by the inplace work.